### PR TITLE
Namespace compatibility changes

### PR DESCRIPTION
**Pull request by newswangerd**
Added:
- `/v3/plugin/ansible/search/namespace-metadata/` -> provides cross repo search functionality for namespace metadata
- `/pulp/api/v3/pulp_ansible/namespaces/` -> API for creating, updating permissions and viewing pulp ansible global namespace

Changed:
- `/v3/namespaces/` -> make the API backwards compatible with galaxy_ng

Jira ticket: https://issues.redhat.com/browse/AAH-2293 (github issue comming soon)

Todo:
- [x] Add filtersets on new APIs
- [x] Tests
- [ ] RBAC
- [ ] v3 compatiblity
    - [ ] error messages
    - [x] filters
    - [ ] allow saving an image with an avatar URL
    - [x] save namespace metadata before launching task to add it to repository

### CODE REVIEW

File: pulp_ansible/app/custom_fields.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
     get_groups_with_perms_attached_roles,
     get_perms_for_model
 )
-from django.contrib.auth.models import Group
+from pulpcore.plugin.models import Group
```
Comment (mdellweg) [inline pulp_ansible/app/custom_fields.py]: I guess, this should have been the proxy model for a long time... Thanks for finding it.

**Comment by kdelee** [comment]
@newswangerd @jctanner what's the status on this? This is blocking some other work afaik to address high CPU usage

**Comment by jctanner** [comment]
@kdelee I think you are looking for https://github.com/pulp/pulp_ansible/pull/1476 instead of this one.

**Review by ShaiahWren — COMMENTED** [review]
A few q's about the metadata api:
- for each listing in ns metadata tab, will there be only ONE ns per repo (e.g., "staging" or "published", or can there be multiple? I am guessing multiple. What unique identifier do you then use for each ns? I'll need some sort of id for editing the metadata, especially if there are multiple namespaces _ibm._ in repository _staging_.

**Comment by ShaiahWren** [comment]
collection upload is looking in the old ns table, please update

**Pull request closed without merging**